go/arrow/array/decimal128.go

```diff
@@ -229,6 +253,67 @@ func (b *Decimal128Builder) newData() (data *Data) {
 	return
 }
 
+func (b *Decimal128Builder) unmarshalOne(dec *json.Decoder) error {
+	t, err := dec.Token()
+	if err != nil {
+		return err
+	}
+
+	var out *big.Float
+
+	switch v := t.(type) {
+	case float64:
+		out = big.NewFloat(v)
+	case string:
+		out, _, err = big.ParseFloat(v, 10, 0, big.ToNearestAway)
```

Comment:
any strong rationale for ToNearestAway?  maybe document?

Reply:
No strong rationale, just happened to be what got me the closest to the value I had input when I used it and I couldn't think of a good way to easily make this configurable in the current interface. I'll add a comment to document it.

Reply:
comments added